controllers/databasecluster_controller.go

```diff
@@ -1485,15 +1488,31 @@ func (r *DatabaseClusterReconciler) reconcilePG(ctx context.Context, req ctrl.Re
 	return r.Status().Update(ctx, database)
 }
 
-func (r *DatabaseClusterReconciler) reconcileLabels(database *everestv1alpha1.DatabaseCluster) {
-	database.ObjectMeta.Labels = map[string]string{
-		databaseClusterNameLabel: database.Name,
+func (r *DatabaseClusterReconciler) reconcileLabels(ctx context.Context, database *everestv1alpha1.DatabaseCluster) error {
+	needsUpdate := false
+	if len(database.ObjectMeta.Labels) == 0 {
+		database.ObjectMeta.Labels = map[string]string{
+			databaseClusterNameLabel: database.Name,
+		}
+		needsUpdate = true
 	}
 	if database.Spec.DataSource != nil {
-		database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.BackupStorageName)] = backupStorageLabelValue
+		if _, ok := database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.BackupStorageName)]; !ok {
+			database.ObjectMeta.Labels[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.BackupStorageName)] = backupStorageLabelValue
+		}
```

Comment:
Looks like if the value was set before it will never be updated. And the unused values may still be present in labels. 
Shouldn't we have the labels up-to-date with what is really used in a DatabaseCluster?